platforms/documentation/docs/src/docs/userguide/best-practices/bp_general.adoc

```diff
@@ -57,3 +57,39 @@ include::sample[dir="snippets/bestPractices/usePluginsBlock/groovy/do-this",file
 
 === Tags
 `#structuring-builds`
+
+[[avoid_duplicate_dependencies]]
+== Avoid Declaring Dependencies Multiple Times
+
+You should avoid duplicating dependencies in your build scripts, whether with different versions or in different dependency configurations.
```

Comment:
Maybe something like this:

```suggestion
Avoid declaring the same dependency multiple times, especially when it is already available transitively or through another configuration.
```

Reply:
I think this is specific to Gradle as Gradle will allow you to declare the same dependency on the same configuration or merged configurations without any warnings.

I think we should highlight that this is about redundant duplication *within* a single build.gradle.kts file, the way it is written it it makes it sound like this is about cross module, and I don't think we should tell people to remove explicit declarations between build.gradle.kts files.

Reply:
I think this could also apply to declaring the same dep on confA and on confB which extendsFrom confA.

Reply:
> the way it is written it it makes it sound like this is about cross module, and I don't think we should tell people to remove explicit declarations between build.gradle.kts files.

Definitely, I'll update the part about transitivity. And I agree that this also applicable to configuration hierarchy.